src/test/java/com/redislabs/redisai/DagTestV2.java

```diff
@@ -0,0 +1,244 @@
+  }
+
+  @Test
+  public void dagExecuteWithoutKeysOrPersist() {
+    setTfModel(client);
+    Dag dag = new Dag();
+    String keyA = "tensorA";
+    String keyB = "tensorB";
+    String keyC = "tensorC";
+    Tensor tA = new Tensor(DataType.FLOAT, new long[] {1, 2}, new float[][] {{1, 2}});
+    Tensor tB = new Tensor(DataType.FLOAT, new long[] {1, 2}, new float[][] {{2, 3}});
+    dag.setTensor(keyA, tA);
+    dag.setTensor(keyB, tB);
+    dag.executeModel("mul", new String[] {keyA, keyB}, new String[] {keyC});
+    dag.getTensor(keyC);
+    List<?> result = client.dagExecute(null, null, new String[] {"mul"}, dag);
+    float[] expected = new float[] {2, 6};
+
+    Tensor tensorC = (Tensor) result.get(3);
+    float[] values = (float[]) tensorC.getValues();
+    Assert.assertEquals("Assert same shape of values", 2, values.length);
+    Assert.assertArrayEquals(expected, values, (float) 0.1);
+  }
+
+  /**
+   * ai.dagExecute simple tensorset modelExecute tensorget positive testing without load or persist
+   * clauses
+   */
+  @Test
+  public void dagExecuteReadOnlyWithoutLoad() {
+    setTfModel(client);
+    Dag dag = new Dag();
+
+    String keyA = "tensorA";
+    String keyB = "tensorB";
+    String keyC = "tensorC";
+    Tensor tA = new Tensor(DataType.FLOAT, new long[] {1, 2}, new float[][] {{1, 2}});
+    Tensor tB = new Tensor(DataType.FLOAT, new long[] {1, 2}, new float[][] {{2, 3}});
+    dag.setTensor(keyA, tA);
+    dag.setTensor(keyB, tB);
+    dag.executeModel("mul", new String[] {keyA, keyB}, new String[] {keyC});
+    dag.getTensor(keyC);
+    List<?> result = client.dagExecuteReadOnly(null, new String[] {"mul"}, dag);
+    float[] expected = new float[] {2, 6};
+    Tensor tensorC = (Tensor) result.get(3);
+    float[] values = (float[]) tensorC.getValues();
+    Assert.assertEquals("Assert same shape of values", 2, values.length);
+    Assert.assertArrayEquals(expected, values, (float) 0.1);
+  }
+
+  @Test
+  // TODO: Last DAG statement with 'NA' response should work after RedisAI fix
+  public void dagExecuteWithExeception() {
+    setTfModel(client);
+    Dag dag = new Dag();
+    String keyA = "tensorA";
+    String keyB = "tensorB";
+    String keyC = "tensorC";
+    Tensor tA = new Tensor(DataType.FLOAT, new long[] {1, 2}, new float[][] {{1, 2}});
+    Tensor tB = new Tensor(DataType.FLOAT, new long[] {2, 3}, new float[][] {{1, 2, 3}, {4, 5, 6}});
+    dag.setTensor(keyA, tA);
+    dag.setTensor(keyB, tB);
+    dag.executeModel("mul", new String[] {keyA, keyB}, new String[] {keyC});
+    //    dag.getTensor(keyC);
+    List<?> result = client.dagExecuteReadOnly(null, new String[] {"mul"}, dag);
+    Assert.assertArrayEquals("OK".getBytes(), (byte[]) result.get(0));
+    Assert.assertArrayEquals("OK".getBytes(), (byte[]) result.get(1));
+    Assert.assertEquals(RedisAIException.class, result.get(2).getClass());
+    //    Assert.assertArrayEquals("NA".getBytes(), (byte[]) result.get(3));
```

Comment:
for the time being, what will `result.get(3)` will return?

Reply:
`NA` as a byte array.

Reply:
@DvirDukhan `NA` as a byte array. Same as OK, two and three lines above.